### DISCUSSION TITLE
Incremental Updates in GraphRAG for Dynamic Datasets

### DISCUSSION THREAD

**Discussion by silhouettehustler** [Azure-Samples/graphrag-accelerator#75] [Q&A] [answered]
Hello GraphRAG Community,

I’ve been using GraphRAG to build knowledge graphs from text data, and it’s been a fantastic tool. However, I’ve encountered a challenge and hope to get some guidance.

My dataset is not static and gets updated regularly. Re-indexing the entire dataset each time there is a small update is not practical due to performance and cost considerations. I’m looking for a way to update only the changed portions of the data without reprocessing everything from scratch.

I’ve explored the documentation and the GraphRAG Solution Accelerator on Azure but haven’t found specific information on handling incremental updates. Has anyone implemented or come across a method for selective re-indexing within GraphRAG?

Any insights, suggestions, or pointers to relevant resources would be greatly appreciated.

Thank you!

**Comment by lucasmtz** [comment]
Did you find an answer @silhouettehustler ?

**Reply by silhouettehustler** [reply, marked as answer]
@lucasmtz not yet, instead opened another discussion on the main [GraphRAG ](https://github.com/microsoft/graphrag) repo.
